src/NServiceBus.Core/Routing/AssemblyRouteSource.cs

```diff
@@ -20,7 +20,7 @@ public AssemblyRouteSource(Assembly messageAssembly, UnicastRoute route)
         public IEnumerable<RouteTableEntry> GenerateRoutes(Conventions conventions)
         {
             var routes = messageAssembly.GetTypes()
-                .Where(t => conventions.IsMessageType(t))
+                .Where(conventions.IsMessageType)
```

Comment:
unless they have changed it, it should allocate a new delegate on each call. Not sure whether that has been changed somehow though. given that there might be quite a few types to filter on, I'd say let's keep the delegate version?